### PR TITLE
Add log file for sql queries - [#5445] for 340

### PR DESCRIPTION
**Pull request by alikon**
Allow the Debug system plugin to log executed queries to a file
#5445 revisited for 340

### CODE REVIEW

File: administrator/language/en-GB/en-GB.plg_system_debug.ini [hunk 1]
```diff
@@ -9,6 +9,8 @@ PLG_DEBUG_ERRORS="Errors"
 PLG_DEBUG_EXPLAIN="Explain"
 PLG_DEBUG_FIELD_ALLOWED_GROUPS_DESC="Optionally restrict users that can see debug information to those in the selected user groups. If none selected, all users will see the debug information."
 PLG_DEBUG_FIELD_ALLOWED_GROUPS_LABEL="Allowed Groups"
+PLG_DEBUG_FIELD_EXECUTEDSQL_DESC="If enabled, executed SQL query will be logged. Only use this setting for short periods of time and for benchmarking purposes."
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_system_debug.ini]: shouldnt this be SQL queries

File: administrator/language/en-GB/en-GB.plg_system_debug.ini [hunk 2]
```diff
@@ -9,6 +9,8 @@ PLG_DEBUG_ERRORS="Errors"
...
+PLG_DEBUG_FIELD_EXECUTEDSQL_LABEL="Log executed SQL"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_system_debug.ini]: Log Executed Queries

**Comment by zero-24** [comment]
@alikon looks cool i have just a a smal CS PR against your branche :smile: https://github.com/alikon/joomla-cms/pull/3

**Comment by Erftralle** [comment]
@test: Works fine for me. :+1:

**Comment by pe7er** [comment]
@test
Patch works ok! 
New option  Plugin Manager: System - Debug "Log Executed Queries" set to YES
+ "Debug System" in Global Configuration set to ON 
= logfile generated with Sitename "Joomla 3.4 test" in file: /logs/Joomla 3.4 test_com_config.sql

Note: the filename contains (IMHO unwanted) spaces. 
Is it necesarry to include the site name in the file name?

**Comment by pe7er** [comment]
I've submitted a PR to your repo to prevent spaces in the file names: https://github.com/alikon/joomla-cms/pull/4

**Comment by brianteeman** [comment]
Thanks for testing - setting RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6202">issues.joomla.org/joomla-cms/6202</a>.</sub>

**Comment by alikon** [comment]
@pe7er 
sitename was an easy way to let me quickly understand from where that log come from

**Comment by pe7er** [comment]
@alikon, Yeah, I understand now. Thanks!

**Comment by msdevcon** [comment]
@Test 
Patch works fine.
Including the "Sitename" in the Filename may not be the best solution, regard Special caracters etc.

**Pull request closed without merging**